Source/FieldSolver/ElectrostaticSolver.cpp

```diff
@@ -62,14 +62,50 @@ WarpX::ComputeSpaceChargeField (bool const reset_fields)
 
     if (do_electrostatic == ElectrostaticSolverAlgo::LabFrame) {
         AddSpaceChargeFieldLabFrame();
-    } else {
+    }
+    else {
+        bool calculate_background_field = false;
         // Loop over the species and add their space-charge contribution to E and B
         for (int ispecies=0; ispecies<mypc->nSpecies(); ispecies++){
             WarpXParticleContainer& species = mypc->GetParticleContainer(ispecies);
             if (species.initialize_self_fields ||
                 (do_electrostatic == ElectrostaticSolverAlgo::Relativistic)) {
                 AddSpaceChargeField(species);
+                calculate_background_field = true;
+            }
+        }
+
+        if (calculate_background_field){
+            // Calculate the E and B components due to boundary potentials alone
+            // Allocate fields for charge and potential
```

Comment:
Could you write a separate function for the code below (similar to `AddSpaceChargeField` and `AddSpaceChargeFieldLabFrame`)? 
This would make the code more "symmetrical" and maybe, in a sense, slightly more readable.